syntax = "proto3";
package monitor.proto;

import "google/protobuf/empty.proto";
import "cpu_load.proto";
import "cpu_stat.proto";
import "cpu_softirq.proto";
import "memory_info.proto";
import "net_info.proto";

message MonitorInfo {
    string name = 1;
    repeated SoftIrq soft_irq = 4;
    CpuLoad cpu_load = 5;
    repeated CpuStat cpu_stat = 6;
    MemoryInfo mem_info = 7;
    repeated NetInfo net_info = 8;
}

service GrpcManager {
    rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty) {
    }
  
    rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo) {
    }
  }
